### ISSUE TITLE
Liveride loses map tiles when switching to another app

### ISSUE THREAD

**Issue by kyegupov** [cyclestreets/android#109]
Steps:
1. start a liveride, wait till map tiles load
2. switch to another app
3. switch back

Observed:
map tiles are lost (can be seen at step 2, just before the switch), only grid and route line are visible. Zooming doesn't help. The only way to fix is to abandon liveride and start it again.

**Comment by yk4ever** [comment]
This happens because of

    getTileProvider().detach();
    getTileProvider().clearTileCache();

lines in `CycleMapView.onPause` called via `LiveRideActivity.onPause`. The corresponding `onResume` method never attaches the provider back.

The usual map browsing (`CycleMapFragment`) is not affected because the fragment, unlike `LiveRideActivity` gets re-created after switching back to app, thus re-initializing the tile providers.

The easiest solution is, probably, to re-create `CycleMapView` on `LiveRideActivity.onPause`.

**Comment by mvl22** [comment]
Thanks for looking at this and diagnosing the problem. (Patches are very welcome!)

**Comment by yk4ever** [comment]
Pull request sent.

**Comment by jezhiggins** [comment]
Thanks.  I'll get on this early next week.

**Comment by yk4ever** [comment]
Any updates?

**Comment by dekarl** [comment]
The patch was merged almost 14 months ago, I guess this issue was simply forgotten?

**Comment by jezhiggins** [comment]
I think you guess right :)
